### PR TITLE
Badpixel

### PR DESCRIPTION
**Pull request by ghost**
Added BadPixelMap class, needed in the step function that corrects for badpixels (developed in branch hsergi) Kevin: can you please review this new Class?

### CODE REVIEW

File: corgidrp/data.py [hunk 1]
```diff
@@ -551,10 +551,49 @@ def __init__(self, data_or_filepath, pri_hdr=None, ext_hdr=None, input_dataset=N
...
+
+        # if this is a new bad pixel map, we need to bookkeep it in the header
+        # b/c of logic in the super.__init__, we just need to check this to see if it is a new bad pixel map
+        if ext_hdr is not None:
+            if input_dataset is None:
+                # error check. this is required in this case
+                raise ValueError("This appears to be a new bad pixel map. The dataset of input files needs to be passed in to the input_dataset keyword to record history of this bad pixel map.")
+            self.ext_hdr['DATATYPE'] = 'BadPixelMap' # corgidrp specific keyword for saving to disk
+
+            # log all the data that went into making this dark
```
Comment (kjl0025) [inline corgidrp/data.py:576]: "dark" --> "bad pixel map"

File: corgidrp/data.py [hunk 2]
```diff
@@ -551,10 +551,49 @@ def __init__(self, data_or_filepath, pri_hdr=None, ext_hdr=None, input_dataset=N
...
+            # add to history
+            self.ext_hdr['HISTORY'] = "Bad Pixel map created"
+
+            # give it a default filename using the first input file as the base
+            # strip off everything starting at .fits
+            orig_input_filename = input_dataset[0].filename.split(".fits")[0]
+            self.filename = "{0}_bad_pixel_map.fits".format(orig_input_filename)
+
+
+        # double check that this is actually a dark file that got read in
```
Comment (kjl0025) [inline corgidrp/data.py:588]: "dark" --> "bad pixel map"

**Review by kjl0025 — CHANGES_REQUESTED** [review]
The guts of this are fine.  I just had a few requests for the docstring and comments.  
Also, go to line 28 of caldb.py and add in your BadPixelMap class.  After those changes, I'm fine with this.

**Comment by hsergi** [comment]
@kjl0025 thanks for all your comments. I have committed a new version. Can you please review it again?

**Review by kjl0025 — APPROVED** [review]
Looks good!  I'll pull it in.

**Pull request merged**